### PR TITLE
Allow the query to still be forwarded when no matching string exists

### PR DESCRIPTION
**Pull request by halfwit**
This is an example of how to implement this, it works in my local testing.
It adds a method, enableDefaultForQuery which can be passed a bool to allow the query to be returned to the final handler, even when it doesn't match an item on the choices list.

### CODE REVIEW

File: extensions/chooser/HSChooser.m [hunk 1]
```diff
@@ -393,7 +394,19 @@ - (void)tableView:(NSTableView *)tableView didClickedRow:(NSInteger)row {
             [skin pushNSObject:choice];
             [skin protectedCallAndError:@"hs.chooser:completionCallback" nargs:1 nresults:0];
         }
-        
+
+        _lua_stackguard_exit(skin.L);
+    } else if (self.enableDefaultForQuery != NO) {
```
Comment (cmsj) [inline extensions/chooser/HSChooser.m:399]: This should also check that `self.completionCallbackRef` is neither `LUA_NOREF` or `LUA_REFNIL`

**Review by cmsj — CHANGES_REQUESTED** [review]
I like the look of this, and I also wonder if it might be worth adding the query string to the `choice` dictionary even when a valid choice exists? That doesn't need to be part of this PR, but we should do the `LUA_NOREF`/`LUA_REFNIL` checks before merging.

**Comment by halfwit** [comment]
Sorry about the commit/revert, not sure what VSCode was doing there.

**Comment by cmsj** [comment]
Thanks! Apologies for the slow merging

**Pull request merged**